backend/internal/userschema/service/userschemaservice.go

```diff
@@ -488,3 +492,394 @@ func logAndReturnServerError(
 	logger.Error(message, log.Error(err))
 	return &constants.ErrorInternalServerError
 }
+
+// ValidateUser validates user attributes against the user schema for the given user type.
+func (us *UserSchemaService) ValidateUser(userType string, userAttributes json.RawMessage) *serviceerror.ServiceError {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, userSchemaLoggerComponentName))
+
+	if userType == "" {
+		logger.Debug("User type is empty, skipping schema validation")
+		return nil
+	}
+
+	if userAttributes == nil {
+		logger.Debug("User has no attributes to validate")
+		return nil
+	}
+
+	if len(userAttributes) == 0 {
+		logger.Debug("User has no attributes to validate")
+		return nil
+	}
+
+	userSchema, err := store.GetUserSchemaByName(userType)
+	if err != nil {
+		if errors.Is(err, constants.ErrUserSchemaNotFound) {
```

Comment:
Is it intentional to allow creating users without a schema? IMO we shouldn't allow this or should at least apply a default schema.